CI: ci-runtime: TestRouterIDAllocation

## CI failure

Failing in about 2% of runs:

```
$ cilium/cilium/operator/pkg/bgpv2$ stress ./bgp.test -test.run=TestRouterIDAllocation
1m0s: 232 runs so far, 5 failures (2.16%), 8 active
```

```
/tmp/go-stress-20250903T120756-936876184
--- FAIL: TestRouterIDAllocation (6.44s)
    --- FAIL: TestRouterIDAllocation/multiple_nodes_and_multiple_instances_with_single_cluster_configs_and_cleanup_the_2_instances_in_the_cluster_config (5.21s)
        logger.go:256: time=2025-09-03T12:08:48.406+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:357 msg="Starting hive"
        logger.go:256: time=2025-09-03T12:08:48.406+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:361 msg="Started hive" duration=31.417µs
        logger.go:256: time=2025-09-03T12:08:48.608+02:00 level=INFO source=/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/manager.go:152 msg="BGPv2 control plane operator started" module=bgp-cp-operator
        cluster_test.go:1578:
            	Error Trace:	/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/cluster_test.go:1590
            	            				/home/raszabo.linux/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.linux-arm64/src/runtime/asm_arm64.s:1268
            	Error:      	"[{test-instance-1 0x40009867f0 0x400069f1b0 <nil> []} {test-instance-2 0x40009867f8 0x400069f1c0 <nil> []} {test-instance-3 0x4000986800 0x400069f1d0 <nil> []}]" should have 2 item(s), but has 3
        cluster_test.go:1578:
            	Error Trace:	/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/cluster_test.go:1578
            	Error:      	Condition never satisfied
            	Test:       	TestRouterIDAllocation/multiple_nodes_and_multiple_instances_with_single_cluster_configs_and_cleanup_the_2_instances_in_the_cluster_config
        logger.go:256: time=2025-09-03T12:08:53.614+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:373 msg="Stopping hive"
        logger.go:256: time=2025-09-03T12:08:53.614+02:00 level=ERROR source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:379 msg="Failed to stop
…
1m0s: 232 runs so far, 5 failures (2.16%), 8 active
1m5s: 254 runs so far, 5 failures (1.97%), 8 active
1m10s: 273 runs so far, 5 failures (1.83%), 8 active
1m15s: 296 runs so far, 5 failures (1.69%), 8 active
1m20s: 312 runs so far, 5 failures (1.60%), 8 active

/tmp/go-stress-20250903T120756-1036067386
--- FAIL: TestRouterIDAllocation (6.42s)
    --- FAIL: TestRouterIDAllocation/single_node_and_cleanup_cluster_config (5.21s)
        logger.go:256: time=2025-09-03T12:09:13.336+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:357 msg="Starting hive"
        logger.go:256: time=2025-09-03T12:09:13.336+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:361 msg="Started hive" duration=31.376µs
        logger.go:256: time=2025-09-03T12:09:13.537+02:00 level=INFO source=/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/manager.go:152 msg="BGPv2 control plane operator started" module=bgp-cp-operator
        cluster_test.go:1578:
            	Error Trace:	/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/cluster_test.go:1583
            	            				/home/raszabo.linux/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.linux-arm64/src/runtime/asm_arm64.s:1268
            	Error:      	"[{{ } {test-node-1     3 0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [{cilium.io/v2 CiliumBGPClusterConfig test-cc-single  0x40001202c0 <nil>}] [] []} {[{test-instance-1 0x40001202c8 0x40007c7e60 <nil> []}]} {[] []}}]" should have 0 item(s), but has 1
        cluster_test.go:1578:
            	Error Trace:	/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/cluster_test.go:1578
            	Error:      	Condition never satisfied
            	Test:       	TestRouterIDAllocation/single_node_and_cleanup_cluster_config
        logger.go:256: time=2025-09-03T12:09:18.543+02:00 level=INFO source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:373 msg="Stopping hive"
        logger.go:256: time=2025-09-03T12:09:18.543+02:00 level=ERROR source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:379 msg="Failed to stop hive" error="context deadline exceeded" duration=24.001µs
FAIL

```

--- a/operator/pkg/bgpv2/cluster_test.go
+++ b/operator/pkg/bgpv2/cluster_test.go
@@ -1568,14 +1568,16 @@ func TestRouterIDAllocation(t *testing.T) {
 				}
 				assert.Equal(c, tt.InitExpectedRouterIDs, InitNodesRouterIDs)
 			}, TestTimeout, 100*time.Millisecond)
-			// cleanup the cluster configs
-			if tt.FinalClusterConfigs != nil {
-				for _, clusterConfig := range tt.FinalClusterConfigs {
-					config := clusterConfig
-					upsertBGPCC(req, ctx, f, config)
-				}
-			}
+
 			assert.EventuallyWithT(t, func(c *assert.CollectT) {
+				// NOTE: upserting cluster configs in "eventually" to workaround race between
+				// resource.Events and resource.Store, where store can contain outdated resource
+				// version even after receiving an event for the new version.
+				if tt.FinalClusterConfigs != nil {
+					for _, clusterConfig := range tt.FinalClusterConfigs {
+						upsertBGPCC(req, ctx, f, clusterConfig)
+					}
+				}
 				NodeConfigs, err := f.bgpnClient.List(ctx, meta_v1.ListOptions{})
 				if !assert.NoError(c, err) {
 					return

